spec/extensions/monads/result_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+RSpec.describe Dry::Types::Result do
+  before { Dry::Types.load_extensions(:monads) }
+
+  let(:type) { Dry::Types['strict.string'] }
+
+  let(:result) { type.try(input) }
+
+  context 'interface' do
+    let(:input) { {} }
+
+    it 'responds to #to_monad' do
+      expect(result).to respond_to(:to_monad)
+    end
+  end
+
+  context 'with valid input' do
+    let(:input) { 'Jane' }
+
+    describe '#to_monad' do
+      it 'wraps Result with Success' do
+        monad = result.to_monad
+
+        expect(monad).to be_a Dry::Monads::Result
+        expect(monad).to be_success
+        expect(monad.value!).to be(result)
```

Comment:
Per my comment re: nesting, this could be `expect(monad.value!).to be(input)`

Reply:
> Per my comment re: nesting, this could be `expect(monad.value!).to be(input)`

Oh, ok, I will do it =)

Reply:
i wonder what @flash-gordon thinks about the idea since it deviates from the example he pointed out in schema does. imo this would be nicer behavior across the board but i recognize that's a much bigger change and modifies existing behavior and needs more thought whether to be a breaking change vs a new method with modified behavior.

Reply:
I think we can use `.to_monad` method and wrap error/input with `Dry::Monads::Result`:
```ruby
result = Dry::Types['strict.string'].try('Jane')
result.class #=> Dry::Types::Success
monad = result.to_monad
monad.class #=> Dry::Monads::Success
result.value == monad.value!  # => true

result = Dry::Types['strict.string'].try(nil)
result.class #=> Dry::Types::Failure
monad = result.to_monad
monad.class #=> Dry::Monads::Failure
result.error == monad.failure  # => true
```

Still this can be problematic... For example in case of extension of `Dry::Types::Result` with some metadata in the feature.